src/optim/FishLeg/fishleg.py

```diff
@@ -105,40 +112,35 @@ def __init__(
         draw: Callable[[nn.Module, torch.Tensor], Tuple[torch.Tensor, torch.Tensor]],
         nll: Callable[[nn.Module, Tuple[torch.Tensor, torch.Tensor]], torch.Tensor],
         aux_dataloader: torch.utils.data.DataLoader,
-        lr: float = 1e-2,
-        eps: float = 1e-4,
+        likelihood: FishLikelihood,
+        fish_lr: float = 5e-2,
         weight_decay: float = 1e-5,
         beta: float = 0.9,
-        update_aux_every: int = -3,
-        aux_lr: float = 1e-3,
+        update_aux_every: int = 10,
+        aux_lr: float = 1e-4,
         aux_betas: Tuple[float, float] = (0.9, 0.999),
         aux_eps: float = 1e-8,
         damping: float = 1e-5,
         pre_aux_training: int = 10,
         differentiable: bool = False,
         sgd_lr: float = 1e-2,
+        initialization: str = 'uniform',
         device: str = "cpu",
+        num_steps = None
     ) -> None:
         self.model = model
-        self.plus_model = copy.deepcopy(self.model)
-        self.minus_model = copy.deepcopy(self.model)
         self.sgd_lr = sgd_lr
-        self.lr = lr
+        self.fish_lr = fish_lr
         self.device = device
+        self.initialization = initialization
 
         self.model = self.init_model_aux(model).to(device)
+        self.likelihood = likelihood
```

Comment:
I think if we are passing in the likelihood anyway, we can probably deprecate the need for `draw` and `nll` to be loaded in?

In the first instance, these can probably just be copied in as methods inside the `FishLeg` class. What do you think?

We can always allow for these methods to be overwritten if needed, but so far I haven't seen any of them look any different.

Reply:
Yep, I think we can rewrite this better by setting default `draw` and `nll`. For language model, they do look a bit different. An example below. 

```
def nll(model, inputs):
    outputs = model(**inputs, head='lm_head')
    pred_y = outputs['logits'][..., :-1, :].contiguous()
    data_y = inputs['labels'][..., 1:].contiguous()
    return -Categorical(
                        logits=pred_y.view(-1, model.config.vocab_size)
                    ).log_prob(data_y.view(-1)).mean()
```